jobs/release-monitoring/discovery/github/Jenkinsfile

```diff
@@ -240,6 +248,39 @@ node {
         isGARelease = hasNewGARelease(componentRelease, latestRelease, productName)
         println "[INFO] latestRelease:${latestRelease}, componentRelease:${componentRelease}, isGARelease:${isGARelease}"
         currentBuild.description = "latest: ${latestRelease}\ncurrent: ${componentRelease}"
+
+        if (releaseTagVar == "") {
+          productVersion = latestRelease
+        }
```

Comment:
Why set this to latestRelease? I see the productVersion and the currentRelease/latestRelease as two separate things, at least for the most part they are different anyway.

Reply:
In some cases, the product version and the release tag are the same ([3Scale](https://github.com/integr8ly/installation/blob/master/evals/inventories/group_vars/all/manifest.yaml#L9), [che](https://github.com/integr8ly/installation/blob/master/evals/inventories/group_vars/all/manifest.yaml#L17)). Would it be best to split these in our manifest file? For each product, there should always be a release version and a product version, even if they have the same value?

Reply:
OK, i understand where you're coming from now. I think what i would like is that we basically use releaseTagVar exactly like what manifestVar had been used for. It is a better representation of what we are passing in i think (the git tag to use). In the case where the releaseTagVar is the same as the product version (che, 3scale etc..), we just don't set productVersionVar, and in the 'Fetch Component Product Version' stage, if that is blank, we set it to the value of the release that was discovered.

This will mean the 2 stages are doing a job each:

'Fetch Latest GitHub Release' - Fetches the latest release based on releaseTagVar and sets latestRelease
'Fetch Component Product Version' - Fetches the latest product version based on productVersionVar and sets productVersion, but if productVersionVar is blank productVersion = latestVersion

Reply:
Can we not just skip this entire stage if "productVersionVar" isn't set?

```
if (productVersionVar) {
   //Do whatever it is we need to do if we have a productVersionVar 
   ...
} else {
   productVersion = latestRelease
}
```